tests/spec/corpus.rs

```diff
@@ -0,0 +1,253 @@
+use std::str::FromStr;
+
+use bson::Bson;
+use pretty_assertions::assert_eq;
+use serde_derive::Deserialize;
+
+use super::run_spec_test;
+
+#[derive(Debug, Deserialize)]
+struct TestFile {
+    description: String,
+    bson_type: String,
```

Comment:
could this be `ElementType`? Then below the skips could compare based on enum cases rather than strings.

Reply:
We can't put it in the struct directly since ElementType doesn't deserialize from strings. While we could parse it in the body of `run_test`, it would be tough to use it for decimal since currently `ElementType::Decimal128Bit` is only defined if the `decimal` feature is enabled, so it would require some boilerplate to define the checks properly.